src/controllers/scripting/legacy/controllerscriptenginelegacy.cpp

```diff
@@ -278,6 +862,92 @@ bool ControllerScriptEngineLegacy::evaluateScriptFile(const QFileInfo& scriptFil
     return true;
 }
 
+#ifdef MIXXX_USE_QML
+std::shared_ptr<QQuickItem> ControllerScriptEngineLegacy::loadQMLFile(
+        const LegacyControllerMapping::ScriptFileInfo& qmlScript,
+        std::shared_ptr<ControllerRenderingEngine> pScreen) {
+    VERIFY_OR_DEBUG_ASSERT(m_pJSEngine ||
+            qmlScript.type !=
+                    LegacyControllerMapping::ScriptFileInfo::Type::Qml) {
+        return nullptr;
+    }
+
+    std::unique_ptr<QQmlComponent>
+            qmlComponent =
+                    std::make_unique<QQmlComponent>(
+                            std::dynamic_pointer_cast<QQmlEngine>(m_pJSEngine).get());
+
+    QFile scene = QFile(qmlScript.file.absoluteFilePath());
+    if (!scene.exists()) {
+        qCWarning(m_logger) << "Unable to load the QML scene:" << qmlScript.file.absoluteFilePath()
+                            << "does not exist.";
+        return nullptr;
+    }
+
```

Comment:
Since in the QML world, everything is a QObject and managed through the QObject-tree (just like it is with widgets), we should embrace the same here IMO. so instead of doing the casting back and forth, ensure that we always create these components with their respective parents, and then use `parented_ptr` (`src/util/parented_ptr.h`) to mark them. Does that make sense?

Reply:
It makes sense but I don't think that works here: `createWithInitialProperties` returns an orphan QObject. It can take a `context` but there is [no indication about ownership](https://doc.qt.io/qt-6.2/qqmlcomponent.html#createWithInitialProperties)
The only component I can apply this is `qmlComponent` but it doesn't live beyond that function, so not sure there is any value. I guess I could move it to stack allocation tho? I kinda feel there was a reason why I didn't but can't remember.